pySDC/implementations/datatype_classes/mesh.py

```diff
@@ -3,6 +3,13 @@
 from pySDC.core.Errors import DataError
 
 try:
+    # Hack in case dolfin is used => must be imported before mpi4py
+    try:
+        import dolfin as df
```

Comment:
Why is this needed here? Or: why wasn't it needed before?

Reply:
At some point I had a test that imported a `fenics_mesh` right after importing the standard `mesh`. Because `mesh` also import `mpi4py` (and also initialize MPI), and `dolfin` does it's own initialization of MPI with C routines, then it produces a call of `MPI_Finalize()` twice.
However, importing `dolfin` before `mpi4py` avoid any clashes of this sort, since `mpi4py` recognizes that MPI has been already initialized before, and uses in the end the same MPI library as `dolfin`.

I spend some time trying to understand that problem, and put this modification in `mesh.py` mostly as a reminder. There is probably a cleaner approach, by handling pySDC imports globaly on the main package level (_i.e_ in `pySDC.__init__.py`), and then let the other sub-modules use packages that have already been imported. But I leave it for latter ... can put a TODO there eventually :wink:

Reply:
Yeah, I was quickly testing problem classes at some point with [`pySDC.tests.test_problem.py`](https://github.com/Parallel-in-Time/pySDC/blob/v6/pySDC/tests/test_problem.py), willing to generalize it for different problem classes.
This requires importing all problem classes on the top of the module (and I added one fenics problem at some point ...). I didn't continue on this way because of the mpi4py/dolfin clash, but I still wanted to include somewhere the safety, because we will probably get this problem again later if we implement more generic/complete testing procedure later

Reply:
And there is later the performance tests for all datatypes => will need to import all datatypes globaly at some point (PS : I'm not very found of local imports, because they can hide problems like this one, which is mostly an ostrich approach)